mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/work/impl/PojoIndexedTypeIndexingPlan.java

```diff
@@ -44,58 +44,58 @@ public PojoIndexedTypeWorkPlan(PojoWorkIndexedTypeContext<I, E, D> typeContext,
 	void add(Object providedId, Object entity) {
 		Supplier<E> entitySupplier = typeContext.toEntitySupplier( sessionContext, entity );
 		I identifier = typeContext.getIdentifierMapping().getIdentifier( providedId, entitySupplier );
-		getWork( identifier ).add( entitySupplier );
+		getPlan( identifier ).add( entitySupplier );
 	}
 
 	@Override
 	void update(Object providedId, Object entity) {
 		Supplier<E> entitySupplier = typeContext.toEntitySupplier( sessionContext, entity );
 		I identifier = typeContext.getIdentifierMapping().getIdentifier( providedId, entitySupplier );
-		getWork( identifier ).update( entitySupplier );
+		getPlan( identifier ).update( entitySupplier );
 	}
 
 	@Override
 	void update(Object providedId, Object entity, String... dirtyPaths) {
 		Supplier<E> entitySupplier = typeContext.toEntitySupplier( sessionContext, entity );
 		I identifier = typeContext.getIdentifierMapping().getIdentifier( providedId, entitySupplier );
-		getWork( identifier ).update( entitySupplier, dirtyPaths );
+		getPlan( identifier ).update( entitySupplier, dirtyPaths );
 	}
 
 	@Override
 	void delete(Object providedId, Object entity) {
 		Supplier<E> entitySupplier = typeContext.toEntitySupplier( sessionContext, entity );
 		I identifier = typeContext.getIdentifierMapping().getIdentifier( providedId, entitySupplier );
-		getWork( identifier ).delete( entitySupplier );
+		getPlan( identifier ).delete( entitySupplier );
 	}
 
 	@Override
 	void purge(Object providedId) {
 		I identifier = typeContext.getIdentifierMapping().getIdentifier( providedId );
-		getWork( identifier ).purge();
+		getPlan( identifier ).purge();
 	}
 
 	void updateBecauseOfContained(Object entity) {
 		Supplier<E> entitySupplier = typeContext.toEntitySupplier( sessionContext, entity );
 		I identifier = typeContext.getIdentifierMapping().getIdentifier( null, entitySupplier );
-		if ( !workPlansPerId.containsKey( identifier ) ) {
-			getWork( identifier ).updateBecauseOfContained( entitySupplier );
+		if ( !indexingPlansPerId.containsKey( identifier ) ) {
+			getPlan( identifier ).updateBecauseOfContained( entitySupplier );
 		}
 		// If the entry is already there, no need for an additional update
 	}
 
 	void resolveDirty(PojoReindexingCollector containingEntityCollector) {
-		for ( IndexedEntityWorkPlan workPerDocument : workPlansPerId.values() ) {
-			workPerDocument.resolveDirty( containingEntityCollector );
+		for ( IndexedEntityIndexingPlan plan : indexingPlansPerId.values() ) {
+			plan.resolveDirty( containingEntityCollector );
 		}
 	}
 
-	void prepare() {
-		sendWorksToDelegate();
+	void process() {
```

Comment:
Is there a reason to use `process` at mapper level and `prepare` at engine/backend level?

Reply:
Well... I didn't want to make this a huge refactoring, so I left out most of the backend side. But since I'm not alone to think the wording in the backend is inconsistent... I added a few commits to address this, and a few more problems.